core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -79,4 +79,8 @@ public Table loadTable(TableIdentifier identifier) {
   protected abstract TableOperations newTableOps(Configuration newConf, TableIdentifier tableIdentifier);
 
   protected abstract String defaultWarehouseLocation(Configuration hadoopConf, TableIdentifier tableIdentifier);
+
+  protected Configuration conf() {
+    return conf;
```

Comment:
I should have caught this before, but this API should not use a Hadoop `Configuration`. This should just be part of the `HiveCatalog`, not the abstract class because all the abstract class does is pass it back to the implementation.